### PR TITLE
feat: test Python 3.12 in CI

### PR DESCRIPTION
**Pull request by texastony**
*Issue #, if available:* Test Python 3.12 in CI.

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

# Check any applicable:
- [ ] Were any files moved? Moving files changes their URL, which breaks all hyperlinks to the files.

### CODE REVIEW

File: .github/workflows/ci_static-analysis.yaml [hunk 1]
```diff
@@ -30,7 +30,7 @@ jobs:
       - uses: actions/checkout@v3
       - uses: actions/setup-python@v4
         with:
-          python-version: 3.8
+          python-version: 3.11
```
Comment (texastony) [inline .github/workflows/ci_static-analysis.yaml:33]:
```suggestion
          python-version: 3.8
```

File: .github/workflows/ci_test-vector-handler.yaml [hunk 2]
```diff
@@ -73,7 +72,7 @@ jobs:
         - uses: actions/checkout@v3
         - uses: actions/setup-python@v4
           with:
-            python-version: 3.9
+            python-version: 3.11
```
Comment (texastony) [inline .github/workflows/ci_test-vector-handler.yaml:75]:
```suggestion
            python-version: 3.9
```

File: tox.ini [hunk 3]
```diff
@@ -1,6 +1,6 @@
 [tox]
 envlist =
-    py{37,38,39,310,311}-{local,integ,accept,examples}, nocmk,
+    py{38,39,310,311,312}-{local,integ,accept,examples}, nocmk,
```
Comment (texastony) [inline tox.ini:3]:
```suggestion
    py{37,38,39,310,311,312}-{local,integ,accept,examples}, nocmk,
```

**Comment by texastony** [comment]
Finally got all CodeBuild targets to start the tests,
as compared to failing to install either Python or the ESDK!

I am fairly certain everything will pass now.

**Review by ajewellamz — APPROVED** [review]
LGTM

**Pull request merged**